Gems/EditorPythonBindings/Code/Source/ActionManager/ActionManagerBus.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/EBus/EBus.h>
+#include <AzCore/Outcome/Outcome.h>
+#include <AzCore/std/string/string.h>
+
+#include <Source/ActionManager/PythonEditorAction.h>
+
+using ActionManagerOperationResult = AZ::Outcome<void, AZStd::string>;
+
+class ActionManagerRequests : public AZ::EBusTraits
+{
+public:
+    static const AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Single;
+
+    //! Register a new Action to the Action Manager.
+    virtual ActionManagerOperationResult RegisterAction(
+        const AZStd::string& contextIdentifier,
+        const AZStd::string& identifier,
+        const AZStd::string& name,
+        const AZStd::string& description,
+        const AZStd::string& category,
+        const AZStd::string& iconPath,
```

Comment:
suggestion: It might be nicer in terms of the API to provide a struct with all these fields and then only pass one parameter. These can all be default initialized or have sensible defaults, whatever makes most sense. It'd be possible to also make builders for them if necessary (or a command style API like we have with the edit/serialize context - and also the manipulator test framework 🙂). This will make the API look a lot cleaner and be helpful if you ever find you need to add an extra field!